src/lib.rs

```diff
@@ -1130,7 +1130,7 @@ impl<EF: ExtraTokenFields, TT: TokenType, TE: ErrorResponseType> Client<EF, TT,
             } else {
                 let error = match serde_json::from_str::<ErrorResponse<TE>>(&reason) {
                     Ok(error) => RequestTokenError::ServerResponse(error),
-                    Err(error) => RequestTokenError::Parse(error),
+                    Err(error) => RequestTokenError::Parse(error, reason.to_string()),
```

Comment:
I ended up completely removing the custom UTF-8 error since that should already be handled in `serde_json::from_slice`. Please tell me what you think about that.